spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/MapTaskExecutionDao.java

```diff
@@ -46,7 +51,67 @@ public void updateTaskExecution(TaskExecution taskExecution) {
 		taskExecutions.put(taskExecution.getExecutionId(), taskExecution);
 	}
 
-	public Map<String, TaskExecution> getTaskExecutions(){
+	@Override
+	public TaskExecution getTaskExecution(String executionId) {
+		return taskExecutions.get(executionId);
+	}
+
+	@Override
+	public long getTaskExecutionCount(String taskName) {
+		int count = 0;
+		for (Map.Entry<String, TaskExecution> entry : taskExecutions.entrySet()) {
+			if (entry.getValue().getTaskName().equals(taskName)) {
+				count++;
+			}
+		}
+		return count;
+	}
+
+	@Override
+	public Set<TaskExecution> findRunningTaskExecutions(String taskName) {
+		Set<TaskExecution> result = new HashSet<>();
+		for (Map.Entry<String, TaskExecution> entry : taskExecutions.entrySet()) {
+			if (entry.getValue().getTaskName().equals(taskName) &&
+					entry.getValue().getEndTime() == null) {
+				result.add(entry.getValue());
+			}
+		}
+		return Collections.unmodifiableSet(result);
```

Comment:
Do we need to wrap this?  The Set is new for each call to the method so it's not like we're exposing the underlying datastore...